### DISCUSSION TITLE
The cuda implementation of  "pyg_lib.sampler.neighbor_sample" function.

### DISCUSSION THREAD

**Discussion by Gibyeng** [pyg-team/pyg-lib#293] [Q&A]
hi,
I want to do some tests on the neighbor sampling process in pyG. But I can not find the implementation of this function  "pyg_lib.sampler.neighbor_sample". I am also confused how to combine cuda c++ code with python and the function is called. Does anyone could help me? thanks a lot!

**Comment by rusty1s** [comment]
You find find the C++ `neighbor_sample` implementation [here](https://github.com/pyg-team/pyg-lib/blob/master/pyg_lib/csrc/sampler/cpu/neighbor_kernel.cpp). Upon installation, the C++ code gets built, and then gets loaded into the Python namespace via the `load_library` function [here](https://github.com/pyg-team/pyg-lib/blob/master/pyg_lib/__init__.py#L22). It is then available under `torch.ops.pyg_lib.neighbor_sample`.

**Reply by Gibyeng** [reply]
Thanks, that helped a lot. 
has the GPU based neighbor_sample not been implemented yet? I can only find GPU based random walk method in .cu flie.

**Reply by rusty1s** [reply]
Yeah, currently we do not have GPU-based neighbor sampling routine.
